packages/core/src/Ogmios/ogmiosToCore/ogmiosToCore.ts

```diff
@@ -0,0 +1,133 @@
+  if (Ogmios.isAlonzoBlock(block)) propName = 'alonzo';
+  if (Ogmios.isBabbageBlock(block)) propName = 'babbage';
+  if (Ogmios.isByronBlock(block)) propName = 'byron';
+  if (Ogmios.isMaryBlock(block)) propName = 'mary';
+  if (Ogmios.isShelleyBlock(block)) propName = 'shelley';
+
+  // If it complains because a branch is not handled, please add logic for the new block type.
+  switch (propName) {
+    case 'allegra':
+      return { block: (block as Ogmios.Schema.Allegra).allegra, kind: 'allegra' };
+    case 'alonzo':
+      return { block: (block as Ogmios.Schema.Alonzo).alonzo, kind: 'alonzo' };
+    case 'babbage':
+      return { block: (block as Ogmios.Schema.Babbage).babbage, kind: 'babbage' };
+    case 'byron':
+      return { block: (block as Ogmios.Schema.Byron).byron, kind: 'byron' };
+    case 'mary':
+      return { block: (block as Ogmios.Schema.Mary).mary, kind: 'mary' };
+    case 'shelley':
+      return { block: (block as Ogmios.Schema.Shelley).shelley, kind: 'shelley' };
+    default: {
+      // will fail at compile time if not all branches are handled
+      // eslint-disable-next-line sonarjs/prefer-immediate-return
+      const _exhaustiveCheck: never = propName;
+      return _exhaustiveCheck;
+    }
+  }
+};
+
+const getBlockBodyHash = (block: Ogmios.Schema.Block): Cardano.BlockId => {
+  const b = getBlockAndKind(block);
+  switch (b.kind) {
+    case 'byron': {
+      return Cardano.BlockId(b.block.hash);
+    }
+    case 'allegra':
+    case 'alonzo':
+    case 'babbage':
+    case 'mary':
+    case 'shelley': {
+      // REVIEW: what to do if blockHash is undefined
+      return Cardano.BlockId(b.block.header?.blockHash || '');
+    }
+    default: {
+      // eslint-disable-next-line sonarjs/prefer-immediate-return
+      const _exhaustiveCheck: never = b;
+      return _exhaustiveCheck;
+    }
+  }
+};
+
+const getSlot = (block: Ogmios.Schema.Block): Cardano.Slot => {
+  const b = getBlockAndKind(block);
+  switch (b.kind) {
+    case 'byron': {
+      if (Ogmios.isByronStandardBlock(block)) {
+        return block.byron.header.slot;
+      }
+      // REVIEW: No slot info for EpochBoundaryBlock. What to do?
+      return 0;
+    }
+    case 'allegra':
+    case 'alonzo':
+    case 'babbage':
+    case 'mary':
+    case 'shelley': {
+      // REVIEW: what to do if slot is undefined
+      return b.block.header?.slot || 0;
+    }
+    default: {
+      // eslint-disable-next-line sonarjs/prefer-immediate-return
+      const _exhaustiveCheck: never = b;
+      return _exhaustiveCheck;
+    }
+  }
+};
+
+/**
+ * Translate `Ogmios` block to `Cardano.BlockMinimal`
+ *
+ * @param ogmiosBlock the block to translate into a `Cardano.BlockMinimal`
+ * @returns {Cardano.BlockMinimal}
+ */
+export const getBlock = (ogmiosBlock: Ogmios.Schema.Block): Cardano.BlockMinimal => {
+  const b = getBlockAndKind(ogmiosBlock);
+  return {
+    header: {
+      blockNo: b.block.header?.blockHeight || 0, // REVIEW: undefined case
```

Comment:
:eyes:

Reply:
https://github.com/input-output-hk/cardano-js-sdk/pull/450#discussion_r976066877

Reply:
So blocks without blockHeight should be ignored. For now let's just return `null` or throw when it's missing.

Reply:
[1e9a079](https://github.com/input-output-hk/cardano-js-sdk/pull/450/commits/1e9a0799abcaf1cad8e6e8f49bee60ec558edc95)